omnigan/utils.py

```diff
@@ -311,14 +319,23 @@ def fake_domains_to_class_tensor(domains):
         ValueError: One of the domains listed is not in {rf, rn, sf, sn}
 
     Returns:
-        torch.Tensor: 1D tensor mapping a domain to an int (not 1-hot)
+        torch.Tensor: 1D tensor mapping a domain to an int (not 1-hot) if loss is CE 
+        or a 2D tensor filled with 0.5 to fool the 
     """
+    if (loss == "l1") or (loss == "l2"):
+        target = torch.FloatTensor(len(domains), 4)
+        target.fill_(0.25)
```

Comment:
0.25 or 0.5?

Reply:
The classifier output 4 logits, with l1 and l2 losses these logits are normalized with a softmax. 
We want an agnostic representation for every domain so  for any image belonging to (rf, rn, sf ,sn) the output probability should be 0.25 for every domain.

Reply:
But don't we want to encode domains like this:

```
rf => [1, 0, 0, 0]
rn => [0, 1, 0, 0]
sf => [0, 0, 1, 0]
sn => [1, 0, 0, 1]
```

And therefore the classifier being completely unsure means it would output 0.5 everywhere?

Other possibility :

```
rf => [1, 0, 1, 0]
rn => [1, 0, 0, 1]
sf => [0, 1, 1, 0]
sn => [0, 1, 0, 1]
```

probably equivalent to 

```
rf => [1, 1]
rn => [1, 0]
sf => [0, 1]
sn => [0, 0]
```